client/src/util/centroid.js

```diff
@@ -0,0 +1,49 @@
+import quantile from "./quantile";
+
+/*
+  Centroid coordinate calculation
+*/
+const calcMeanCentroid = (world, annoName, annoValue, layoutDimNames) => {
+  const centroid = [0, 0, 0];
+  const annoArray = world.obsAnnotations.col(annoName).asArray();
+  const layoutXArray = world.obsLayout.col(layoutDimNames[0]).asArray();
+  const layoutYArray = world.obsLayout.col(layoutDimNames[1]).asArray();
+
+  for (let i = 0, len = annoArray.length; i < len; i += 1) {
+    if (annoArray[i] === annoValue) {
+      centroid[0] += layoutXArray[i];
+      centroid[1] += layoutYArray[i];
+      centroid[2] += 1;
+    }
+  }
+
+  if (centroid[2] !== 0) {
```

Comment:
Are you handling if the count is 0? This shouldn't happen, but it looks like if it did, then it would be in the 0,0 xy position.

Reply:
I only handled to prevent division by zero.  I didn't make a conscious choice on how to draw that centroid. How this is handled is probably a question for @sidneymbell and @colinmegill

Reply:
I believe that we pull the categories + labels (i.e., `annoValue`s) directly from the data. This means that if a label exists, then there is at least one cell that has that label.

The one place I could see this potentially getting into trouble would be if all of those cells are somehow assigned a nonfinite value (i.e., `NaN`, `Inf` or `-Inf`) for their x and/or y coordinates.
None of the layouts from scanpy should return non-finite coordinates, but if someone's passed a custom embedding, it could happen. 

@colinmegill / @csweaver  - how do we currently handle cells with non-finite embedding coordinates? 

The tl;dr -- it would be an extreme edge case, but if all of the cells with a given `annoValue` have non-finite values for their layout coordinates, this will currently implode. I'd suggest adding a quick conditional to check and just not showing a label in that case.

Reply:
Made it so no centroids are drawn if there are no finite values in the category